### DISCUSSION TITLE
Alternative operations inside in a routing

### DISCUSSION THREAD

**Discussion by matteocarcano** [frePPLe/frepple#651] [Q&A] [answered]
Hi,
How can i model some alternative operations that are parte of a routing?

I tried to set the following schema:

Operation A - type routing
Operation B - type alternate - owner operation A
Operation C - type time_per - owner operation B
Operation D - type time_per - owner operation B

But it doesn't work because i can't set routing operation as alternate's owner. Only time_per or fixed_time can have owners. 

How can i do? 

Thank you in advance.

**Comment by jdetaeye** [comment]
You can't use a structure as you describe.   

The steps in a routing need to be of type "time_per" or "fixed-time".
If there are choices for a routing step you can use on of the following approaches:
- use a aggregate resource to model a case like "operation B is done on resource X, which has subresources Y and Z".
- use a alternate resources to model a case like "operation B is done on resource Y or resource Z"
- use alternate materials to model a case like "operation B consumes item D or item F"

If your case isn't one of the above three, you need to have a top alternate operation with 3 different routings A1, A2 and A3.

**Reply by matteocarcano** [reply]
Sorry I'm not sure i have understood.

In my model i need to have the following schema:

Operation A (first routing operation) : 
  - i have to  done it on small resource if the quantity is small
  - i have to done it on big resource if the quantity is big.
 
Operation B (second routing operation) that is a time_per operation without issues.

What's better way to do it?

Thank you

**Reply by jdetaeye** [reply, marked as answer]
Just define 2 different routings to produce the same item:
- routing "make X using small resource", size-minimum=1, size-maximum=100, priority=1, item=X
      - time per operation "make X using small resource - A" 
      - ... other steps 
- routing "make X using big resource", size-minimum=100, priority=2, item=X
     - time per operation "make X using big resource - A" 
     - ... other steps 

Frepple automatically puts an alternate operation on top of these 2 routings.

**Reply by matteocarcano** [reply]
Thank you very much
